src/core/ddsi/src/ddsi_config.c

```diff
@@ -1177,6 +1184,30 @@ static enum update_result uf_tracingOutputFileName (struct cfgst *cfgst, UNUSED_
   return URES_SUCCESS;
 }
 
+static enum update_result uf_random_seed (struct cfgst *cfgst, void *parent, struct cfgelem const * const cfgelem, UNUSED_ARG (int first), const char *value)
+{
+  ddsrt_prng_seed_t * const elem = cfg_address(cfgst, parent, cfgelem);
+  if (strcmp(value, "") == 0) {
+    ddsrt_prng_makeseed(elem);
+  } else {
+    ddsrt_md5_byte_t buf[16];
+    ddsrt_md5_state_t md5st;
+    ddsrt_md5_init (&md5st);
+    ddsrt_md5_append (&md5st, (ddsrt_md5_byte_t *) value, (unsigned int) strlen(value));
+    ddsrt_md5_finish (&md5st, buf);
+    memcpy((ddsrt_md5_byte_t*)elem, buf, 16);
+    memcpy(((ddsrt_md5_byte_t*)elem) + 16, buf, 16);
+  }
+  return URES_SUCCESS;
+}
+
+static void pf_random_seed (struct cfgst *cfgst, void *parent, struct cfgelem const * const cfgelem, uint32_t sources)
+{
+  ddsrt_prng_seed_t * const seed = cfg_address (cfgst, parent, cfgelem);
+  cfg_logelem (cfgst, sources, "%"PRIu32" %"PRIu32" %"PRIu32" %"PRIu32" %"PRIu32" %"PRIu32" %"PRIu32" %"PRIu32"",
```

Comment:
So if I configure the seed to be "triskaidecafobie" here it'll print `5722008f 768a9428 306e67eb e5014210 5722008f 768a9428 306e67eb e5014210` on a big-endian machine and `8f002257 28948a76 eb676e30 104201e5 8f002257 28948a76 eb676e30 104201e5` on a little-endian machine (if I didn't get it wrong).

I have no idea whether the two result in the same sequence (I don't know the Mersenne Twister that well), and it would be difficult to reproduce it based on the configuration dump. But arguably the former matters not and the latter could even be considered a feature.

Just noting it in case some of this is important to you but you forgot it earlier 😀

Reply:
I did indeed consider that using the same seed on little or big endian machines would most likely result in different names. However, going to the effort of ensuring equivalent outcomes between hosts is actually what you do *not* want in all cases that I can come up with. In fact, I am still on the fence about something we previously discussed: should we *always* include the hostname in the seed? What about *process/executable name*? Maybe some config booleans, e.g. `<EntityAutoNaming seed="hexakosioihexekontahexafobie" mixin_host="true" mixin_executable="true">fancy</EntityAutoNaming>`

Reply:
Ah, it appears you run into scary integers later than I do if start enumerating them from, let's say, 0. (If I'd say 14, there'd the possibility I'd never run into a scary one anymore, so that'd cheating, no? 🙂)

It is probably a good idea to make mixing in host name and executable information an option. That way we can keep sitting on the fence.